app/controllers/posts_controller.rb

```diff
@@ -3,7 +3,7 @@ class PostsController < ApplicationController
 
   def index
     @post = Post.new
-    timeline_posts
+    @timeline_posts = @post.timeline_posts(current_user)
   end
 
```

Comment:
- [ ] PostsController#index displays posts in a timeline (their own posts + posts from their friends), but your project is not working like so. When user a sends a request to user b and user b accepts the requests, both users should be able to see each other posts but it is not working like that presently, only the user that accepts the friend request is able to see the posts of the user they accepted their request whereas it should be both ways i.e a mutual relationship. I am not sure why this is happening because your query in the post model for fetching the posts seems to be ok. I suspect it is because of the name you gave the method. The collision with the name in the post controller could be a possible reason and also I don't know why you are doing `@post.timeline_post(current_user)` and not `Post.timeline_posts(current_user).
- [ ] You can easily solve this issue by having a simple method like this in your user model
```ruby
def friends_and_own_posts
  Post.where(user: (self.friends.clone << self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end
```
and simply calling it in the post controller as `User.friends_and_own_posts` and you should have the post of both the current user and their friends.

Reply:
I already implemented most of the things you mentioned here. It seems you did not take a closer look at my project. The mutual friendship aspect is what I have corrected so that the timeline is working perfectly now in my own view.